infra/src/main/java/com/redhat/service/bridge/infra/exceptions/ErrorsService.java

```diff
@@ -0,0 +1,15 @@
+package com.redhat.service.bridge.infra.exceptions;
+
+import java.util.Optional;
+
+import com.redhat.service.bridge.infra.models.ListResult;
+import com.redhat.service.bridge.infra.models.QueryInfo;
+
+public interface ErrorsService {
+
+    ListResult<Error> getErrors(QueryInfo pageInfo);
+
+    Optional<Error> getError(int errorId);
+
+    Optional<Error> getError(Exception e);
```

Comment:
Not sure it feels right that we can't always return an Error given an `Exception`. That means that our API is going to be missing error/errors codes for certain failures. It feels like we need a default `Error` that will be returned unless there is an `Error` specifically configured for the passed `Exception`.

It looks like that is the functionality of the current implementation, so happy to do that in another JIRA if you agree?

Reply:
Returning optional is just a convenience thing so the layer invoking this handle the situation in which there is no error for an exception. Logic that Im not sure belongs to ErrorService.